From 65a5bb5c8a0515769651facb8aa1ed16e7d7e365 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sun, 3 Mar 2024 20:39:29 +0200
Subject: [PATCH] Clear all neighbor entry items explicitly

Do not leave some of the variables to their previously used values when
a neighbor entry is cleared.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 src/ap/neighbor_db.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ap/neighbor_db.c b/src/ap/neighbor_db.c
index b0480ffea..d6ddd3d1c 100644
--- a/src/ap/neighbor_db.c
+++ b/src/ap/neighbor_db.c
@@ -99,7 +99,10 @@ static void hostapd_neighbor_clear_entry(struct hostapd_neighbor_entry *nr)
 	nr->civic = NULL;
 	os_memset(nr->bssid, 0, sizeof(nr->bssid));
 	os_memset(&nr->ssid, 0, sizeof(nr->ssid));
+	os_memset(&nr->lci_date, 0, sizeof(nr->lci_date));
 	nr->stationary = 0;
+	nr->short_ssid = 0;
+	nr->bss_parameters = 0;
 }
 
 
-- 
2.43.0

